### ISSUE TITLE
Gate.io perpetual - Delay on retrieving the data of the open position resulting to the client to recognize the incorrect amount

### ISSUE THREAD

**Issue by nikspz** [hummingbot/hummingbot#6032]
1. There is a delay on gateio perp on retrieving the data of the open position resulting to the client to recognize the incorrect amount 
2. Issue happened when order to OPEN postion filled, after bot returns `ERROR - Unexpected error in user stream listener loop. (AttributeError: 'GateIoPerpetualDerivative' object has no attribute '_account_positions`
3. This also could be reproduced when stop/restart the client (With position opened)

Note: `Bot was able to recover after few minutes`

**Steps To Reproduce**
1. Create a perpetualMM strategy 
2. Set tight spreads to Open the position
3. Wait till order to OPEN position will be filled
4. Bot returns `ERROR - Unexpected error in user stream listener loop. (AttributeError: 'GateIoPerpetualDerivative' object has no attribute '_account_positions')`
5. Check status and logs and review order to CLOSE the position

Actual:
Delay on retrieving the data of the open position resulting to the client to recognize the incorrect amount.
Bot was able to recover after few minutes and place order with correct amount

Expected:
Bot able to retrieving the data of the open position.
Bot able to place order to CLOSE with correct amount

[logs_gateper.log](https://github.com/hummingbot/hummingbot/files/10496340/logs_gateper.log)
[gateper.zip](https://github.com/hummingbot/hummingbot/files/10496339/gateper.zip)
```
2023-01-24 13:30:53,090 - 3600847 - hummingbot.connector.client_order_tracker - INFO - Created LIMIT BUY order t-HBOTBBCUT5f3028658c630116954 for 0.0005 to OPEN a BTC-USDT position.
2023-01-24 13:30:53,099 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567053.0, "type": "OrderType.LIMIT", "trading_pair": "BTC-USDT", "amount": "0.0005", "price": "22939.9", "order_id": "t-HBOTBBCUT5f3028658c630116954", "creation_timestamp": 1674567053.0, "exchange_order_id": "248102164781", "leverage": 1, "position": "OPEN", "event_name": "BuyOrderCreatedEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:30:53,406 - 3600847 - hummingbot.connector.client_order_tracker - INFO - Created LIMIT SELL order t-HBOTSBCUT5f3028658c71e116954 for 0.0005 to OPEN a BTC-USDT position.
2023-01-24 13:30:53,416 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567053.0, "type": "OrderType.LIMIT", "trading_pair": "BTC-USDT", "amount": "0.0005", "price": "22962.9", "order_id": "t-HBOTSBCUT5f3028658c71e116954", "creation_timestamp": 1674567053.0, "exchange_order_id": "248102166288", "leverage": 1, "position": "OPEN", "event_name": "SellOrderCreatedEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:31:51,174 - 3600847 - hummingbot.connector.client_order_tracker - INFO - The BUY order t-HBOTBBCUT5f3028658c630116954 amounting to 0.0005/0.0005 BTC has been filled.
2023-01-24 13:31:51,191 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Maker buy order of 0.0005 BTC filled. [clock=2023-01-24 13:31:51+00:00]
2023-01-24 13:31:51,191 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567111.0, "order_id": "t-HBOTBBCUT5f3028658c630116954", "trading_pair": "BTC-USDT", "trade_type": "TradeType.BUY", "order_type": "OrderType.LIMIT", "price": "22939.9", "amount": "0.0005", "trade_fee": {"percent": "0", "percent_token": "USDT", "flat_fees": [{"token": "USDT", "amount": "0.00172049250000000006864675494711036662920378148555755615234375"}]}, "exchange_trade_id": "155742101", "leverage": 1, "position": "OPEN", "event_name": "OrderFilledEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:31:51,192 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Maker buy order t-HBOTBBCUT5f3028658c630116954 (0.0005 BTC @ 22939.9 USDT) has been completely filled. [clock=2023-01-24 13:31:51+00:00]
2023-01-24 13:31:51,200 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567111.0, "order_id": "t-HBOTBBCUT5f3028658c630116954", "base_asset": "BTC", "quote_asset": "USDT", "base_asset_amount": "0.0005", "quote_asset_amount": "11.46995", "order_type": "OrderType.LIMIT", "exchange_order_id": "248102164781", "event_name": "BuyOrderCompletedEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:31:51,201 - 3600847 - hummingbot.connector.client_order_tracker - INFO - BUY order t-HBOTBBCUT5f3028658c630116954 completely filled.
2023-01-24 13:31:52,000 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Canceling the limit order t-HBOTSBCUT5f3028658c71e116954. [clock=2023-01-24 13:31:52+00:00]
2023-01-24 13:31:52,001 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Initiated cancelation of sell order t-HBOTSBCUT5f3028658c71e116954 in favour of take profit order.
2023-01-24 13:31:52,129 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567112.0, "order_id": "t-HBOTSBCUT5f3028658c71e116954", "exchange_order_id": "248102166288", "event_name": "OrderCancelledEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:31:52,129 - 3600847 - hummingbot.connector.client_order_tracker - INFO - Successfully canceled order t-HBOTSBCUT5f3028658c71e116954.
2023-01-24 13:34:38,000 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Creating stop loss sell order to close long position.
2023-01-24 13:34:38,001 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Creating 1  LIMIT ask orders at (Size, Price): ['0.0005 BTC, 22768.1 USDT'] to CLOSE position.
2023-01-24 13:34:38,156 - 3600847 - hummingbot.connector.derivative.gate_io_perpetual.gate_io_perpetual_derivative.GateIoPerpetualDerivative - ERROR - Unexpected error in user stream listener loop.
Traceback (most recent call last):
  File "/home/ubuntu/staging/hummingbot/connector/derivative/gate_io_perpetual/gate_io_perpetual_derivative.py", line 551, in _user_stream_event_listener
    await self._process_account_position_message(position_msg)
  File "/home/ubuntu/staging/hummingbot/connector/derivative/gate_io_perpetual/gate_io_perpetual_derivative.py", line 593, in _process_account_position_message
    del self._account_positions[pos_key]
AttributeError: 'GateIoPerpetualDerivative' object has no attribute '_account_positions'
2023-01-24 13:34:38,219 - 3600847 - hummingbot.connector.client_order_tracker - INFO - Created LIMIT SELL order t-HBOTSBCUT5f30293c1fccf116954 for 0.0005 to CLOSE a BTC-USDT position.
2023-01-24 13:34:38,229 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567278.0, "type": "OrderType.LIMIT", "trading_pair": "BTC-USDT", "amount": "0.0005", "price": "22768.1", "order_id": "t-HBOTSBCUT5f30293c1fccf116954", "creation_timestamp": 1674567278.0, "exchange_order_id": "248103463420", "leverage": 1, "position": "CLOSE", "event_name": "SellOrderCreatedEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:34:43,157 - 3600847 - hummingbot.connector.client_order_tracker - INFO - The SELL order t-HBOTSBCUT5f30293c1fccf116954 amounting to 0.0005/0.0005 BTC has been filled.
2023-01-24 13:34:43,175 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Maker sell order of 0.0005 BTC filled. [clock=2023-01-24 13:34:43+00:00]
2023-01-24 13:34:43,175 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567283.0, "order_id": "t-HBOTSBCUT5f30293c1fccf116954", "trading_pair": "BTC-USDT", "trade_type": "TradeType.SELL", "order_type": "OrderType.LIMIT", "price": "22868", "amount": "0.0005", "trade_fee": {"percent": "0", "percent_token": "USDT", "flat_fees": [{"token": "USDT", "amount": "0.0057169999999999998596678096873802132904529571533203125"}]}, "exchange_trade_id": "155742912", "leverage": 1, "position": "CLOSE", "event_name": "OrderFilledEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:34:43,176 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Maker sell order t-HBOTSBCUT5f30293c1fccf116954 (0.0005 BTC @ 22768.1 USDT) has been completely filled. [clock=2023-01-24 13:34:43+00:00]
2023-01-24 13:34:43,185 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567283.0, "order_id": "t-HBOTSBCUT5f30293c1fccf116954", "base_asset": "BTC", "quote_asset": "USDT", "base_asset_amount": "0.0005", "quote_asset_amount": "11.4340", "order_type": "OrderType.LIMIT", "exchange_order_id": "248103463420", "event_name": "SellOrderCompletedEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:34:43,185 - 3600847 - hummingbot.connector.client_order_tracker - INFO - SELL order t-HBOTSBCUT5f30293c1fccf116954 completely filled.
2023-01-24 13:34:44,001 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Creating stop loss sell order to close long position.
2023-01-24 13:34:45,000 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Creating stop loss sell order to close long position.
2023-01-24 13:34:49,001 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Creating stop loss sell order to close long position.
2023-01-24 13:34:50,000 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Creating stop loss sell order to close long position.
................
2023-01-24 13:35:36,001 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Creating stop loss sell order to close long position.
2023-01-24 13:35:37,001 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Creating stop loss sell order to close long position.
2023-01-24 13:35:38,001 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Creating stop loss sell order to close long position.
2023-01-24 13:35:38,001 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Creating 1  LIMIT ask orders at (Size, Price): ['0.0005 BTC, 22768.1 USDT'] to CLOSE position.
2023-01-24 13:35:38,121 - 3600847 - hummingbot.connector.client_order_tracker - INFO - The SELL order t-HBOTSBCUT5f3029755856b116954 amounting to 0.0005/0.0005 BTC has been filled.
2023-01-24 13:35:38,137 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Maker sell order of 0.0005 BTC filled. [clock=2023-01-24 13:35:38+00:00]
2023-01-24 13:35:38,137 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567338.0, "order_id": "t-HBOTSBCUT5f3029755856b116954", "trading_pair": "BTC-USDT", "trade_type": "TradeType.SELL", "order_type": "OrderType.LIMIT", "price": "22879", "amount": "0.0005", "trade_fee": {"percent": "0", "percent_token": "USDT", "flat_fees": [{"token": "USDT", "amount": "0.005719750000000000007827072323607353609986603260040283203125"}]}, "exchange_trade_id": "155743387", "leverage": 1, "position": "CLOSE", "event_name": "OrderFilledEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:35:38,138 - 3600847 - hummingbot.connector.client_order_tracker - INFO - Created LIMIT SELL order t-HBOTSBCUT5f3029755856b116954 for 0.0005 to CLOSE a BTC-USDT position.
2023-01-24 13:35:38,147 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567338.0, "type": "OrderType.LIMIT", "trading_pair": "BTC-USDT", "amount": "0.0005", "price": "22768.1", "order_id": "t-HBOTSBCUT5f3029755856b116954", "creation_timestamp": 1674567338.0, "exchange_order_id": "248103991136", "leverage": 1, "position": "CLOSE", "event_name": "SellOrderCreatedEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:35:38,148 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Maker sell order t-HBOTSBCUT5f3029755856b116954 (0.0005 BTC @ 22768.1 USDT) has been completely filled. [clock=2023-01-24 13:35:38+00:00]
2023-01-24 13:35:38,158 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567338.0, "order_id": "t-HBOTSBCUT5f3029755856b116954", "base_asset": "BTC", "quote_asset": "USDT", "base_asset_amount": "0.0005", "quote_asset_amount": "11.4395", "order_type": "OrderType.LIMIT", "exchange_order_id": "248103991136", "event_name": "SellOrderCompletedEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:35:38,158 - 3600847 - hummingbot.connector.client_order_tracker - INFO - SELL order t-HBOTSBCUT5f3029755856b116954 completely filled.
2023-01-24 13:35:47,002 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Creating 1 LIMIT bid orders at (Size, Price): ['5 BTC, 22833.2 USDT'] to CLOSE position.
2023-01-24 13:35:47,092 - 3600847 - hummingbot.connector.derivative.gate_io_perpetual.gate_io_perpetual_derivative.GateIoPerpetualDerivative - NETWORK - Error submitting buy LIMIT order to Gate_io_perpetual for 5.0000 BTC-USDT 22833.2.
Traceback (most recent call last):
  File "/home/ubuntu/staging/hummingbot/connector/exchange_py_base.py", line 481, in _create_order
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/ubuntu/staging/hummingbot/connector/derivative/gate_io_perpetual/gate_io_perpetual_derivative.py", line 302, in _place_order
    order_result = await self._api_post(
  File "/home/ubuntu/staging/hummingbot/connector/exchange_py_base.py", line 858, in _api_post
    return await self._api_request(*args, **kwargs)
  File "/home/ubuntu/staging/hummingbot/connector/exchange_py_base.py", line 896, in _api_request
    request_result = await rest_assistant.execute_request(
  File "/home/ubuntu/staging/hummingbot/core/web_assistant/rest_assistant.py", line 75, in execute_request
    raise IOError(f"Error executing request {method.name} {url}. HTTP status is {response.status}. "
OSError: Error executing request POST https://api.gateio.ws/api/v4/futures/usdt/orders. HTTP status is 400. Error: {"label":"INSUFFICIENT_AVAILABLE","detail":"margin 114325.8152751 while available 32.69428944802"}

2023-01-24 13:35:47,092 - 3600847 - hummingbot.connector.derivative.gate_io_perpetual.gate_io_perpetual_derivative.GateIoPerpetualDerivative - WARNING - Failed to submit buy order to Gate_io_perpetual. Check API key and network connection.
2023-01-24 13:35:47,094 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567347.0, "order_id": "t-HBOTBBCUT5f30297dedcc1116954", "order_type": "OrderType.LIMIT", "event_name": "MarketOrderFailureEvent", "event_source": "gate_io_perpetual"}
2023-01-24 13:35:47,095 - 3600847 - hummingbot.connector.client_order_tracker - INFO - Order t-HBOTBBCUT5f30297dedcc1116954 has failed. Order Update: OrderUpdate(trading_pair='BTC-USDT', update_timestamp=1674567347.0, new_state=<OrderState.FAILED: 6>, client_order_id='t-HBOTBBCUT5f30297dedcc1116954', exchange_order_id=None, misc_updates=None)
2023-01-24 13:36:10,316 - 3600847 - hummingbot.connector.derivative.gate_io_perpetual.gate_io_perpetual_derivative.GateIoPerpetualDerivative - WARNING - Failed to fetch trade updates for order t-HBOTBBCUT5f30297dedcc1116954. Error: Skipped order update with order fills for t-HBOTBBCUT5f30297dedcc1116954 - waiting for exchange order id.
2023-01-24 13:37:09,001 - 3600847 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (BTC-USDT) Creating 1 LIMIT bid orders at (Size, Price): ['0.0005 BTC, 22833.2 USDT'] to CLOSE position.
2023-01-24 13:37:09,081 - 3600847 - hummingbot.connector.client_order_tracker - INFO - Created LIMIT BUY order t-HBOTBBCUT5f3029cc211a9116954 for 0.0005 to CLOSE a BTC-USDT position.
2023-01-24 13:37:09,090 - 3600847 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1674567429.0, "type": "OrderType.LIMIT", "trading_pair": "BTC-USDT", "amount": "0.0005", "price": "22833.2", "order_id": "t-HBOTBBCUT5f3029cc211a9116954", "creation_timestamp": 1674567429.0, "exchange_order_id": "248104580408", "leverage": 1, "position": "CLOSE", "event_name": "BuyOrderCreatedEvent", "event_source": "gate_io_perpetual"}
```

**Screenshots**

gate.io perpetual, Bot returns `ERROR - Unexpected error in user stream listener loop. (AttributeError: 'GateIoPerpetualDerivative' object has no attribute '_account_positions')`
then it tried to create order with amount of 5 BTC

![image](https://user-images.githubusercontent.com/83953535/214486984-9d343364-49b1-43bb-84cd-8ba7aa3655d6.png)

I restarted the bot and it saw the position with amount of 5 BTC

![image](https://user-images.githubusercontent.com/83953535/214487089-b875c02e-eb8c-4190-8730-b1ca33d6f07b.png)

Same issue for different pair (from 30 XRP to 3 XRP)

![image](https://user-images.githubusercontent.com/83953535/214487676-0f90bd04-6cfd-41c3-96a1-ff2739b50aff.png)

**Release version**
dev-1.12.0

**Comment by nikspz** [comment]
Severity: P3
Bounty: 10000 (https://hummingbot.org/maintenance/bugs/)
Rationale: Bug related to Gate.io Perpetual, Silver tier connector

**Comment by MementoRC** [comment]
It looks like `self._account_positions` should be `self._perpetual_trading._account_positions`. It's an attribute of `PerpetualTrading`, not `PerpetualDerivativePyBase`

**Comment by yancong001** [comment]
I would like to be assigned this ticket

**Comment by nikspz** [comment]
We took time to test both PRs and decided to assign to @yancong001 since his PR seems closer to solve the issue.

**Comment by nikspz** [comment]
Closing this ticket as Paid.
Fix has been deployed to development branch and will be available on the next release version 1.13.0.
Thank you for your contribution
